### PR TITLE
Updates based on feature/rename output validity proof

### PR DESCRIPTION
**Pull request by nicocartesi**
Updates to docs based on https://github.com/cartesi/rollups/pull/122/files

### CODE REVIEW

File: cartesi-rollups/api/graphql/objects/output-validity-proof.mdx [hunk 1]
```diff
@@ -9,21 +9,21 @@ Validity proof for an output
 
 ```graphql
 type OutputValidityProof {
-  inputIndex: Int!
-  outputIndex: Int!
+  inputIndexWithinEpoch: Int!
+  outputIndexWithinInput: Int!
```
Comment (gligneul) [inline cartesi-rollups/api/graphql/objects/output-validity-proof.mdx:13]: You missed the change in line 31.

File: cartesi-rollups/api/graphql/queries/notice.mdx [hunk 2]
```diff
@@ -10,7 +10,7 @@ Get notice based on its index
 ```graphql
 notice(
   noticeIndex: Int!
-  inputIndex: Int!
+  inputIndexWithinEpoch: Int!
```
Comment (gligneul) [inline cartesi-rollups/api/graphql/queries/notice.mdx:13]: This field didn't change. The only field that changed was the one in the OutputValidityProof.

File: cartesi-rollups/api/graphql/queries/report.mdx [hunk 3]
```diff
@@ -10,7 +10,7 @@ Get report based on its index
 ```graphql
 report(
   reportIndex: Int!
-  inputIndex: Int!
+  inputIndexWithinEpoch: Int!
```
Comment (gligneul) [inline cartesi-rollups/api/graphql/queries/report.mdx:13]: This didn't change.

File: cartesi-rollups/api/graphql/queries/voucher.mdx [hunk 4]
```diff
@@ -10,7 +10,7 @@ Get voucher based on its index
 ```graphql
 voucher(
   voucherIndex: Int!
-  inputIndex: Int!
+  inputIndexWithinEpoch: Int!
```
Comment (gligneul) [inline cartesi-rollups/api/graphql/queries/voucher.mdx:13]: This didn't change.

File: cartesi-rollups/api/json-rpc/sol-output.md [hunk 5]
```diff
@@ -241,7 +241,7 @@ struct OutputValidityProof {
   bytes32 vouchersEpochRootHash;
   bytes32 noticesEpochRootHash;
   bytes32 machineStateHash;
-  bytes32[] keccakInHashesSiblings;
+  bytes32[] outputHashInOutputHashesSiblings;
```
Comment (gligneul) [inline cartesi-rollups/api/json-rpc/sol-output.md:244]: Missed changes in inputIndex and outputIndex.

File: cartesi-rollups/api/rollup.yaml [hunk 6]
```diff
@@ -290,7 +290,7 @@ components:
           format: uint64
           description: Deprecated. Always receives 0.
           example: 0
-        input_index:
```
Comment (gligneul) [inline cartesi-rollups/api/rollup.yaml:293]: This didn't change.

**Comment by nicocartesi** [comment]
Hi @gligneul , thanks a lot for the feedback! Can you please check if everything has been addressed correctly?

File: cartesi-rollups/api/json-rpc/sol-output.md (hunk 5 above)
Comment (gligneul) [reply cartesi-rollups/api/json-rpc/sol-output.md:244]: Still missing changes. Lines 238 and 239.

File: cartesi-rollups/api/json-rpc/sol-output.md [hunk 7]
```diff
@@ -250,11 +250,11 @@ struct OutputValidityProof {
 
 | Name                        | Type      | Description                                                       |
 | --------------------------- | --------- | ----------------------------------------------------------------- |
-| inputIndex                 | uint256   | Which input, inside the epoch, the output belongs to                                |
-| outputIndex                 | uint256   | Index of output emitted by the input                      |
+| inputIndexWithinEpoch                 | uint256   | Which input, inside the epoch, the output belongs to                                |
+| outputIndexWithinEpoch                 | uint256   | Index of output emitted by the input                      |
```
Comment (gligneul) [inline cartesi-rollups/api/json-rpc/sol-output.md:254]:
```suggestion
| outputIndexWithinInput                 | uint256   | Index of output emitted by the input                      |
```

File: cartesi-rollups/api/typeDefs.graphql [hunk 8]
```diff
@@ -71,11 +71,11 @@ type Query {
...
   input(index: Int!): Input!
   "Get voucher based on its index"
-  voucher(voucherIndex: Int!, inputIndex: Int!): Voucher!
+  voucher(voucherIndex: Int!, inputIndexWithinEpoch: Int!): Voucher!
   "Get notice based on its index"
-  notice(noticeIndex: Int!, inputIndex: Int!): Notice!
+  notice(noticeIndex: Int!, inputIndexWithinEpoch: Int!): Notice!
   "Get report based on its index"
-  report(reportIndex: Int!, inputIndex: Int!): Report!
+  report(reportIndex: Int!, inputIndexWithinEpoch: Int!): Report!
```
Comment (gligneul) [inline cartesi-rollups/api/typeDefs.graphql:78]: These didn't change. Use the following link as a reference https://github.com/cartesi/rollups/blob/main/offchain/graphql-server/schema.graphql

File: cartesi-rollups/api/rollup/finish.api.mdx [hunk 9]
```diff
@@ -25,7 +25,7 @@ When the Rollup HTTP Server returns 202, the DApp backend should retry the call
...
... status field for the first finish request and after an inspect-state request.\n\nIf the advance-state request is rejected, the vouchers and notices are discarded.\nIn contrast, reports are not discarded in case of rejection.\nWhen running inside a Cartesi machine, the Cartesi Server Manager reverts the entire state of the machine to what it was before receiving the request.\n\nDuring a finish call, the next rollup request might not be immediately available.\nWhen the DApp backend and the Rollup HTTP Server are running inside a Cartesi machine, the Cartesi Server Manager pauses the whole machine execution until the next request is ready.\nWhen running in host mode, the Rollup HTTP Server returns the status code 202 after 10 seconds to avoid the connection timing out.\nWhen the Rollup HTTP Server returns 202, the DApp backend should retry the call to finish passing the same arguments as before.\n","type":"text/plain"},"url":{"path":["finish"],"host":["{{baseUrl}}"],"query":[],"variable":[]},"header":[{"key":"Content-Type","value":"application/json"},{"key":"Accept","value":"application/json"}],"method":"POST","body":{"mode":"raw","raw":"\"\"","options":{"raw":{"language":"json"}}}}}
```
Comment (gligneul) [inline cartesi-rollups/api/rollup/finish.api.mdx:28]: There was no change in this API. The diff is hard to read, but this seems wrong.

File: cartesi-rollups/api/rollup/finish.api.mdx [hunk 10]
```diff
@@ -69,7 +69,7 @@ When the Rollup HTTP Server returns 202, the DApp backend should retry the call
...
...} defaultValue={undefined}></SchemaItem></TabItem><TabItem label={"MOD2"} value={"1-item-properties"}><SchemaItem collapsible={false} name={"payload"} required={false} schemaDescription={"The payload is in the Ethereum hex binary format.\nThe first two characters are '0x' followed by pairs of hexadecimal numbers that correspond to one byte.\nFor instance, '0xdeadbeef' corresponds to a payload with length 4 and bytes 222, 173, 190, 175.\nAn empty payload is represented by the string '0x'.\n"} schemaName={"string"} qualifierMessage={undefined} defaultValue={undefined}></SchemaItem></TabItem></SchemaTabs></div></div></div></details></SchemaItem></ul></details></TabItem><TabItem label={"Example (from schema)"} value={"Example (from schema)"}><ResponseSamples responseExample={"{\n  \"request_type\": \"advance_state\",\n  \"data\": {\n    \"metadata\": {\n      \"msg_sender\": \"string\",\n      \"epoch_index\": 0,\n      \"input_index_within_epoch\": 0,\n      \"block_number\": 0,\n      \"timestamp\": 0\n    },\n    \"payload\": \"string\"\n  }\n}"} language={"json"}></ResponseSamples></TabItem></SchemaTabs></TabItem></MimeTabs></div></TabItem><TabItem label={"202"} value={"202"}><div>
```
Comment (gligneul) [inline cartesi-rollups/api/rollup/finish.api.mdx:72]: >  There was no change in this API. The diff is hard to read, but this seems wrong.

Here as well.

**Review by gligneul — APPROVED** [review]
LGTM

**Comment by gligneul** [comment]
I think you should squash all these commits

**Comment by claudioantonio** [comment]
@tuler (cc. @nicocartesi ) could you please merge this PR to `next`?

**Pull request merged**